### DISCUSSION TITLE
Is there a simulation example?

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [nasa/fprime#1945] [Q&A]
Is there an example of a ready-made FPrime application that runs in a 2D or 3D simulated environment?

I know if the Ref application and the Math tutorial. I am now looking for something that is a "visual" demo.

Thanks

**Comment by LeStarch** [comment]
I don't know of any demos like that.  From the question, I am taking it to mean that you are looking for a demo that runs a system (robot, spacecraft, etc.) with simulated visual 2D/3D output.  As far as I know, we don't have such a thing.

You might look into generic system simulations and running F´ software on the simulated CPU.

**Comment by SterlingPeet** [comment]
Just as a contextual FYI: I did recently see a presentation about [NOS3](https://github.com/nasa/nos3) that indicated _their_ project was intending to include F Prime as a FSW option.  This would leverage Goddard's [42](https://github.com/ericstoneking/42) simulator as the frontend.

**Reply by WiilyLin** [reply]
Actually, they write FSW by CFE. Therefore, if we want to use the simulator in NOS3. Which part is possible to fix in F'?
